contracts/minter-imago/src/contract.rs

```diff
@@ -0,0 +1,743 @@
+    if env.block.time >= config.start_time {
+        return Err(ContractError::AlreadyStarted {});
+    }
+
+    // If current time already passed the new start_time return error
+    if env.block.time > start_time {
+        return Err(ContractError::InvalidStartTime(start_time, env.block.time));
+    }
+
+    let genesis_start_time = Timestamp::from_nanos(GENESIS_MINT_START_TIME);
+    // If the new start_time is before genesis start time return error
+    if start_time < genesis_start_time {
+        return Err(ContractError::BeforeGenesisTime {});
+    }
+
+    config.start_time = start_time;
+    CONFIG.save(deps.storage, &config)?;
+    Ok(Response::new()
+        .add_attribute("action", "update_start_time")
+        .add_attribute("sender", info.sender)
+        .add_attribute("start_time", start_time.to_string()))
+}
+
+pub fn execute_update_per_address_limit(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    per_address_limit: u32,
+) -> Result<Response, ContractError> {
+    let mut config = CONFIG.load(deps.storage)?;
+    if info.sender != config.admin {
+        return Err(ContractError::Unauthorized(
+            "Sender is not an admin".to_owned(),
+        ));
+    }
+    if per_address_limit == 0 || per_address_limit > MAX_PER_ADDRESS_LIMIT {
+        return Err(ContractError::InvalidPerAddressLimit {
+            max: MAX_PER_ADDRESS_LIMIT,
+            min: 1,
+            got: per_address_limit,
+        });
+    }
+    config.per_address_limit = per_address_limit;
+    CONFIG.save(deps.storage, &config)?;
+    Ok(Response::new()
+        .add_attribute("action", "update_per_address_limit")
+        .add_attribute("sender", info.sender)
+        .add_attribute("limit", per_address_limit.to_string()))
+}
+
+// if admin_no_fee => no fee,
+// else if in whitelist => whitelist price
+// else => config unit price
+pub fn mint_price(deps: Deps, is_admin: bool) -> Result<Coin, StdError> {
+    let config = CONFIG.load(deps.storage)?;
+
+    if is_admin {
+        return Ok(coin(AIRDROP_MINT_PRICE, config.unit_price.denom));
+    }
+
+    if config.whitelist.is_none() {
+        return Ok(config.unit_price);
+    }
+
+    let whitelist = config.whitelist.unwrap();
+
+    let wl_config: WhitelistConfigResponse = deps
+        .querier
+        .query_wasm_smart(whitelist, &WhitelistQueryMsg::Config {})?;
+
+    if wl_config.is_active {
+        Ok(wl_config.unit_price)
+    } else {
+        Ok(config.unit_price)
+    }
+}
+
+fn mint_count(deps: Deps, info: &MessageInfo) -> Result<u32, StdError> {
+    let mint_count = (MINTER_ADDRS
+        .key(info.sender.clone())
+        .may_load(deps.storage)?)
+        .unwrap_or(0);
+    Ok(mint_count)
+}
+
+fn last_block_minted(deps: Deps, info: &MessageInfo) -> Result<u64, StdError> {
+    let last_block_minted = (MINTER_LAST_BLOCK
+        .key(info.sender.clone())
+        .may_load(deps.storage)?)
+        .unwrap_or(0);
+    Ok(last_block_minted)
+}
```

Comment:
I wish we had transient stores available in contracts for this specific use case but this works!